ydb/docs/en/core/deploy/configuration/config.md

```diff
@@ -456,6 +456,60 @@ actor_system_config:
 | `resolution` | When making a schedule for sending messages, discrete time slots are used. The slot duration is set by the `resolution` parameter in microseconds. |
 
+## Memory {#memory}
+
+Several components utilize memory, including but not limited to:
+
+- Shared Cache: Stores recently accessed data pages read from Blob Storage to reduce disk I/O and accelerate data retrieval;
+- MemTable: Holds data that has not yet been flushed to SST;
+- KQP: Executes queries and stores their intermediate results;
+- Allocator Caches: Keeps memory blocks which have been released but not yet returned to the operating system.
+
+Memory limits can be configured to control the overall memory usage, ensuring the database operates efficiently within the available resources.
+
+### Hard memory limit
+
+The hard memory limit specifies the total amount of available memory.
+
+By default, the hard memory limit is set to the process's CGroup memory limit. 
+
+In environments without a CGroup memory limit, the default hard memory limit is equal to the total available memory of the host. This allows the database to utilize all available resources in unrestricted environments, though it may lead to resource contention with other processes running on the same host.
+
+Additionally, the hard memory limit can be specified in the configuration. Note that the database process may still exceed this limit, so it is highly recommended to use CGroup memory limits in production environments to enforce strict memory control.
+
+Example of the `memory_controller_config` section with a specified hard memory limit:
+
+```yaml
+memory_controller_config:
+  hard_limit_bytes: 16106127360
+```
+
+### Per component memory limits
+
+Certain components have their own memory limits. Currently, these components are Shared Cache and MemTable.
+
+Most memory limits have both minimum and maximum thresholds, allowing for dynamic adjustment based on current process consumption.
+
+Memory limits can be configured either in absolute bytes or as a percentage relative to the hard memory limit. Using percentages is advantageous for managing database clusters with nodes of varying capacities. If both absolute byte limits and percentage limits are specified, the system uses a combination of both.
+
+Example of the `memory_controller_config` section with specified Shared Cache limits:
+
+```yaml
+memory_controller_config:
+  shared_cache_min_percent: 10
+  shared_cache_max_percent: 30
+```
+
+Parameters | Description | Default
+--- | --- | ---
+`hard_limit_bytes` | A hard memory usage limit for the database. | CGroup&nbsp;memory&nbsp;limit&nbsp;/<br/>Host memory
```

Comment:
Even though it is not technically correct, "a/an/the" at the beginning of such description tables is often dropped for brevity. It might help with the table width here, too.